jest.config.js

```diff
@@ -1,4 +1,6 @@
 module.exports = {
   preset: 'ts-jest',
   testEnvironment: 'node',
+  restoreMocks: true,
+  resetMocks: true,
```

Comment:
>`restore` is a superset of `reset`, which is a superset of `clear`. See:

I thought the same thing! But apparently `restoreAllMocks` is not a superset of `resetAllMocks`. I discovered this in testing. My best guess as to why is that `restoreAllMocks` will reset and restore all mocks that can be restored, but it won't reset those that aren't able to be restored (this is what I was referring to here: https://consensys.slack.com/archives/G1L7H42BT/p1619037207210500?thread_ts=1618992261.203300&cid=G1L7H42BT)

Reply:
I did get tests passing on this branch. I didn't see any remaining reliance upon non-reset mocks :thinking: 

>Sometimes it's convenient to set up a mock implementation once, and keep that implementation for the duration the test suite.

It's convenient but it's _dangerous_, because it hides dependencies between tests. I have yet to find a case where defining behavior in `beforeEach` instead of `before` is more difficult or inconvenient.

Even resetting mocks doesn't go quite far enough in my opinion - we should be avoiding using shared mocks _at all_, and reset/restore them just as a safeguard. That would go even farther to ensure test isolation (though of course it still would not be a guarantee).